### ISSUE TITLE
Yesod devel InternalError - “devel.hs: signal: 15”

### ISSUE THREAD

**Issue by bhyland** [yesodweb/yesod#1650]
I have a yesod app that compiles and runs fine on my main dev machine (Ubuntu 16.04). Yesterday I cloned the repo on another machine running Ubuntu 18.04. 

It compiles fine, however when I try to run in devel mode it compiles but shows the 'Application isn't built' page at localhost:3000. When I press enter to recompile I get an error like so:
```
devel.hs: runFakeHandler issue: InternalError "devel.hs: signal: 15"
CallStack (from HasCallStack):
  error, called at src/Yesod/Core/Unsafe.hs:24:14 in yesod-core-1.6.14-8UP8zoy9zgT7ImCVVtHxpr:Yesod.Core.Unsafe
Success! Waiting for next file change.
Type help for available commands. Press enter to force a rebuild.
Starting devel application
Devel application launched: http://localhost:3000
Made application.
```
As you can see, it then says the build was successful, and waits for changes. localhost:3000 still shows the 'application isn't built' page.

The compiled app works correctly when I call it e.g. with
```
stack exec my-app -- settings.yml
```
so this seems to be specifically about yesod devel. 

So far I have tried:

- removing and reinstalling latest stack, yesod-bin etc.
- deleting ~/.stack and .stack-work and recompiling everything
- checked that libs such as postgres-dev, sqlite3, zlib1g-dev, etc. exist

I have also asked on SO to no avail. Can anyone give me any clues as to what this error means and what to try next?

Thanks for any help!

----
* System
   -   Ubuntu 18.04 running kernal 4.15.0-72-generic x64
   -   stack 2.1.3
   -   yesod-bin 1.6.0.3
   -   ghc 8.6.4

* Dependencies:
   - https://gist.github.com/bhyland/7a955d3cb1aa8b3612fd0029fbad6f30

* Is there anything custom or unusual about your setup? 
   - No

**Comment by snoyberg** [comment]
Can you check if the same thing happens when using the plain scaffolding?

**Comment by bhyland** [comment]
Thanks for getting back to me! Here's what I tried last night:

1. Plain Scaffolding

I set up a new project with ```stack new yesodweb/sqlite3```.  Running ```stack exec yesod -- devel``` gave the same ```devel.hs: Signal 15``` error:

```
Yesod devel server. Enter 'quit' or hit Ctrl-C to quit.
Application can be accessed at:

http://localhost:3000
https://localhost:3443
If you wish to test https capabilities, you should set the following variable:
  export APPROOT=https://localhost:3443

Success! Waiting for next file change.
Type help for available commands. Press enter to force a rebuild.
Starting devel application
Devel application launched: http://localhost:3000

devel.hs: devel.hs: signal: 15
Success! Waiting for next file change.
Type help for available commands. Press enter to force a rebuild.
Starting devel application
Devel application launched: http://localhost:3000
```
As before, although build succeeded, the app did not show up on localhost. Pressing enter to rebuild gave the ```devel.hs: signal 15``` error again (though notice the text around it is different this time). 

However this time the app did show up on localhost, and seems to be working correctly. Also note that the scaffolded site downloaded a newer version of ghc (8.6.5).

2. Updating to later LTS.

I then tried updating my application to LTS-14.19 and running ```yesod devel```. 

Result: Same error as in the initial post, no localhost access:

```
...
Registering library for rtr-api-1.0.385..
devel.hs: runFakeHandler issue: InternalError "devel.hs: signal: 15"
CallStack (from HasCallStack):
  error, called at src/Yesod/Core/Unsafe.hs:24:14 in yesod-core-1.6.17-Ia58cz6VLaYHQXmnGEgwef:Yesod.Core.Unsafe
Success! Waiting for next file change.
Type help for available commands. Press enter to force a rebuild.
Starting devel application
Devel application launched: http://localhost:3000
Made application.
Checking for db defaults
Database User inserts complete
```

**Comment by snoyberg** [comment]
I'm not sure where the error is coming from then, sorry. My best recommendation is to try adding some debug statements inside the `getApplicationDev` function in `src/Application.hs` and see if you can determine where exactly that function is erroring.

**Comment by bhyland** [comment]
I had a little time to get back to this issue today, to no avail.

Unfortunately getApplicationDev (and everything else from the app/devel.hs entry point onwards) happens after that error - the error is reported immediately after the 'Registering library' message and  just before --success-hook is called. 

I've been through the yesod-bin code and I'm not really getting anywhere - I can't even figure out why runFakeHandler is even being called, since it's only supposed to be used in Yesod test???

Anyway, since I can work around it, unless there are any further hints people can give me I'll close this issue as just one of those mysterious things that happen when you upgrade ubuntu too many times.

**Comment by bhyland** [comment]
I finally figured out what was going on here.

Basically I had set the sqlite poolsize to 1 while trying to troubleshoot something else, and had not set it back. Setting the sqlite poolsize back to 10 fixed it.

I have some code that populates defaults in the database after migration, which of course only runs on the creation of a new database. This was where yesod devel was hanging before it was ready to serve, since the inserts were never finishing (for whatever reason).

The tricky part (other than no error messages from sqlite coming through) was that this looked as though it was working correctly on the other systems, since the sqlite database was already set up on those machines and these inserts were never running.
